page/renderer/ComponentUtils.js

```diff
@@ -0,0 +1,67 @@
+export default class ComponentUtils {
+  /**
+   * Initializes the registry used for managing component utils.
+   *
+   * @param {ObjectContainer} oc The application's dependency injector - the
+   *        object container.
+   */
+  constructor(oc) {
+    /**
+     * The application's dependency injector - the object container.
+     *
+     * @type {ObjectContainer}
+     */
+    this._oc = oc;
+
+    /**
+     * Map of registered utilities.
+     *
+     * @type {Object<string, Object>}
+     */
+    this._utilities = {};
+  }
+
+  /**
+   * Registers single utility class or multiple classes in alias->class mapping.
+   *
+   * @param {function(new: Object)|Object<string, function(new: Object)>} componentUtilityClass
+   * @param {string|null} alias
+   * @returns {Object|Object<string, Object>}
+   */
+  register(componentUtilityClass, alias = null) {
+    if (typeof componentUtilityClass === 'function') {
+      this._utilities[
+        alias || componentUtilityClass.name
```

Comment:
The componentUtilityClass.name can be tricky with scrambling class name

Reply:
Removed this feature. Changed `alias` to `name` and made it mandatory.